include/bluetooth/audio/csis.h

```diff
@@ -345,15 +343,21 @@ int bt_csis_client_release_set(void);
 void bt_csis_client_register_cb(struct bt_csis_client_cb *cb);
 
 /**
- * @brief Read the lock value of a specific device and instance.
+ * @brief Check if an array of set members are unlocked
+ *
+ * This will read the set lock value on all members and respond with a single
+ * state.
  *
- * @param conn      Pointer to the connection to the device.
- * @param inst_idx  Index of the CSIS index of the peer device (as it may have
- *                  multiple CSIS instances).
+ * @param members   Array of set members to check lock state for.
+ * @param count     Number of set members in @p members.
+ * @param set       Pointer to the specified set, as a member may be part of
+ *                  multiple sets.
  *
  * @return Return 0 on success, or an errno value on error.
  */
-int bt_csis_client_lock_get(struct bt_conn *conn, uint8_t inst_idx);
+int bt_csis_client_get_lock_state(const struct bt_csis_client_set_member **members,
+				  uint8_t count,
+				  const struct bt_csis_client_set *set);
 
 /**
  * @brief Lock an array of set members
```

Comment:
This may be a suitable place to try to explain the problems I have with naming.
To me, an "array of set members" (like it is used in this function) sounds like a set (a number of devices that belong together).
However, the set members each contain an "array of sets", and a set (of that kind) is really a "set membership" (?)
I'll probably get it with time, but I think it is possible to find better naming.

Could the lowest level (bt_csis_client_set) be called e.g. or "set_info" to indicate that this is information about a set the device is a member of (rather than a set)?
The next level (_set_member) could be kept as is (I believe this is the information for one device, containing information about the device is a member of).
The top level (array of set members) could then be called a set.

Reply:
Or possibly the lowest lever could be a "set_membership"?
Then a set would be an array of "set_member"s, which would each have a number of "set_memberships".

Reply:
> It would also help if we could (here) use "set" only for the noun, and not for the verb.

Yeah, it would be best to avoid any form of `set_set` :D 

> To me, an "array of set members" (like it is used in this function) sounds like a set (a number of devices that belong together).

They are similar, but not the same. 
A set may consist of 3 devices {SM_1, SM_2, SM_3}, but an array of set members could be a single set member [SM_1] or some [SM_1, SM_2] or all [SM_1, SM_2, SM_3]. 

Likewise, there may also be two sets: {SM_1, SM_2} and {SM_2, SM_3}, i.e. SM_2 is part of multiple sets, and as such must have multiple CSIS instances.  

I agree that `bt_csis_client_set` is not the optimal naming, as it really a representation of a remote CSIS instance, and not a set.  `bt_csis_client_set_info` seem to be a better choice indeed, but it is also contains more than information, so that is not optimal either in my opinion. Perhaps something like `bt_csis_client_inst`?

Reply:
I see your point that the array may be only a partial set.  So it may not be correct to call the array for a set.

As for the second point, I see that my comment was lacking some words - I agree that at set_member (?) may be a member of several sets.

Your proposal for naming (`bt_csis_client_inst`) is an improvement over the current naming - I would be satisfied with that.

Reply:
I can live with that - we have seen worse before.  I find it better than calling the CSIS-instance a "set".

TheCSIS instance contains the SIRK, the set size and the rank of a device (plus an internal member), and there is one such instance for each set the device is a member of.  It could also possibly be called a "membership".  (In which case a set member will end up with several memberships, which seems OK.)

Reply:
I'll update the PR with `bt_csis_client_csis_inst`. 

If we ever rename the API to follow the roles instead, it would become something like `bt_set_coordinator_csis_inst` or possibly `bt_set_coord_csis_inst` which would avoid the douple `csis`.